### PR TITLE
feat(plugin): add button link field component

### PR DESCRIPTION
**Pull request by nina992**
# Overview
 implement the field component Button Link
# Screenshot

## Memo
Design Link : https://www.figma.com/file/xWM8xrtIAh5oMj0seLd9Jd/view-2.0?node-id=2415%3A82836

### CODE REVIEW

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts [hunk 1]
```diff
@@ -132,11 +134,22 @@ type PointStroke = {
   }[];
 };
 
+export type ButtonLinkInfo = {
+  title?: string;
+  link?: string;
+};
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:140]: I don't think you need to separate this in to its own type. For the ButtonLink, we don't have multiple items or anything, so having these types directly on the ButtonLink type I think is sufficient.

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export const fieldName = {
   camera: "カメラ",
   legend: "凡例",
+  buttonLink: "Button link",
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:4]: リンクボタン

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/index.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { ComponentType } from "react";
 
 // general
+import ButtonLink from "./general/ButtonLink";
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/index.ts:4]: It is really small, but can you try to keep order the same (at least in the same file). It is just a small thing but will allow our brains to focus less on finding things and more on other things.
So the // general import order = `fields` order = FieldComponent type order etc.

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        <FieldTitle>Link</FieldTitle>
+        <FieldValue>
+          <TextInput onChange={handleChangeButtonLink} />
+        </FieldValue>
+      </Field>
+    </Wrapper>
+  ) : (
+    <Wrapper>
+      <Field>
+        <FieldTitle />
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx:41]: No FieldTitle for the non-editMode?

Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx:41]: Don't need this, right?

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        <FieldValue>
+          <StyledButton onClick={() => window.open(CurrentButton?.link, "_blank", "noopener")}>
+            <Text>{CurrentButton?.title}</Text>
+          </StyledButton>
+        </FieldValue>
+      </Field>
+    </Wrapper>
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx:48]: Only the StyledButton is necessary, right?

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts [hunk 6]
```diff
@@ -132,11 +134,22 @@ type PointStroke = {
   }[];
 };
 
+export type ButtonLinkInfo = {
+  title?: string;
+  link?: string;
+};
+
+export type ButtonLink = FieldBase<"buttonLink"> & {
+  type: "buttonLink";
```
[the remark this answers is not in the document]
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:143]: done

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts (hunk 1 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:140]: done

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts (hunk 2 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:4]: done

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx (hunk 5 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx:48]: I will only remove the field title to keep same design

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/index.ts (hunk 3 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/index.ts:4]: sorry I didn't get te idea , as some times especially in the import part , it is auto order , do you mean in type definition

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts]: 30294f5aa82bcbb0ea19d6d665ee117a42cea95a

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts (hunk 1 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts]: 30294f5aa82bcbb0ea19d6d665ee117a42cea95a

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts (hunk 2 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts]: 30294f5aa82bcbb0ea19d6d665ee117a42cea95a

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/index.ts (hunk 3 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/index.ts:4]: 30294f5aa82bcbb0ea19d6d665ee117a42cea95a

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx (hunk 5 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx]: 30294f5aa82bcbb0ea19d6d665ee117a42cea95a

**Review by KaWaite — APPROVED** [review]

**Pull request merged**